code/lib/cli/src/utils.ts

```diff
@@ -15,3 +20,65 @@ export function getEnvConfig(program: Record<string, any>, configEnv: Record<str
     }
   });
 }
+
+/**
+ * Given a file name, creates an object with utilities to manage a log file.
+ * It creates a temporary log file which you can manage with the returned functions.
+ * You can then decide whether to move the log file to the users project, or remove it.
+ *
+ * @example
+ * ```
+ *  const { logStream, moveLogFile, removeLogFile, clearLogFile, readLogFile } = await createLogStream('my-log-file.log');
+ *
+ *  // SCENARIO 1:
+ *  // you can write custom messages to generate a log file
+ *  logStream.write('my log message');
+ *  await moveLogFile();
+ *
+ *  // SCENARIO 2:
+ *  // or you can pass it to stdio and capture the output of that command
+ *  try {
+ *    await this.executeCommand({
+ *      command: 'pnpm',
+ *      args: ['info', packageName, ...args],
+ *      // do not output to the user, and send stdio and stderr to log file
+ *      stdio: ['ignore', logStream, logStream]
+ *    });
+ *  } catch (err) {
+ *    // do something with the log file content
+ *    const output = await readLogFile();
+ *    // move the log file to the users project
+ *    await moveLogFile();
+ *  }
+ *  // success, no need to keep the log file
+ *  await removeLogFile();
+ *
+ * ```
+ */
+export const createLogStream = async (
+  logFileName: string
+): Promise<{
+  moveLogFile: () => Promise<void>;
+  removeLogFile: () => Promise<void>;
+  clearLogFile: () => Promise<void>;
+  readLogFile: () => Promise<string>;
+  logStream: WriteStream;
+}> => {
+  const finalLogPath = join(process.cwd(), logFileName);
+  const temporaryLogPath = tempy.file({ name: logFileName });
+
+  const logStream = createWriteStream(temporaryLogPath, { encoding: 'utf8' });
```

Comment:
Currently, the stream is written directly to the hard drive. Users would be thankful if this operation could be avoided to avoid multiple write operations to the hard drive in a short period of time. The hard drive will thank it as well ;) Instead, should we consider writing into memory instead, until the file is finally written? This has several benefits:
 
1. Write operations to the hard drive are slow in comparison to the memory
2. There is no need to clear the temporary file if error logs are not written to the hard drive; just clear the memory.
3. Reading from the temporary file can be avoided completely because we don't need it anymore

Let's just write to the hard drive, when it is necessary (producing the final log file), otherwise, let's avoid it as far as possible.

Reply:
Good point! I'll try it out